src/Data/Singletons/TypeLits/Internal.hs

```diff
@@ -183,6 +185,24 @@ infixr 8 %^
 -- Defunctionalization symbols for type-level (^)
 $(genDefunSymbols [''(^)])
 
+-- | The singleton analogue of 'TN.<=?' 
+--
+-- Note that, because of historical reasons in GHC's 'TN.Nat' API, 'TN.<=?'
+-- is incompatible (unification-wise) with '<=' and the 'PEq', 'SEq',
```

Comment:
It's worth clarifying that when you refer to `(<=)` here, you're referring to the promoted `(<=)` family from `PEq` (and not the [`(<=)` from `GHC.TypeNats`](https://hackage.haskell.org/package/base-4.10.1.0/docs/GHC-TypeNats.html#t:-60--61-), since that's defined as `type (<=) x y = (x <=? y) ~ True`).